File: python/reverse-string/reverse_string_test.py
import unittest

from reverse_string import (
    reverse,
)

# Tests adapted from `problem-specifications//canonical-data.json`


class ReverseStringTest(unittest.TestCase):
    def test_an_empty_string(self):
        self.assertEqual(reverse(""), "")

    def test_a_word(self):
        self.assertEqual(reverse("robot"), "tobor")

    def test_a_capitalized_word(self):
        self.assertEqual(reverse("Ramen"), "nemaR")

    def test_a_sentence_with_punctuation(self):
        self.assertEqual(reverse("I'm hungry!"), "!yrgnuh m'I")

    def test_a_palindrome(self):
        self.assertEqual(reverse("racecar"), "racecar")

    def test_an_even_sized_word(self):
        self.assertEqual(reverse("drawer"), "reward")


if __name__ == "__main__":
    unittest.main()

File: typescript/reverse-string/reverse-string.test.ts
import { reverse } from "./reverse-string";

describe("Reverse String", () => {
  it("an empty string", () => {
    const expected = "";
    expect(reverse("")).toEqual(expected);
  });

  it("a word", () => {
    const expected = "tobor";
    expect(reverse("robot")).toEqual(expected);
  });

  it("a capitalized word", () => {
    const expected = "nemaR";
    expect(reverse("Ramen")).toEqual(expected);
  });

  it("a sentence with punctuation", () => {
    const expected = `!yrgnuh m'I`;
    expect(reverse(`I'm hungry!`)).toEqual(expected);
  });

  it("a palindrome", () => {
    const expected = "racecar";
    expect(reverse("racecar")).toEqual(expected);
  });
});
